README.md

```diff
@@ -6,63 +6,139 @@
-| container\_port | Postgres port | number | 5432 | no |
-| container\_image | Postgres docker image | string | "postgres:12-alpine" | no |
-| admin\_user | Postgres admin username | string or data obj | "postgres" | yes |
-| admin\_password | Postgres admin password | string or data obj | "postgres" | yes |
-| admin\_password | Postgres database name | string | "metastore" | yes |
-| container\_environment\_variables | Postgres container environement variables | list(string) | ["PGDATA=/var/lib/postgresql/data"] | no |
-| volume\_destination | Postgres volume destination | string | "/var/lib/postgresql/data" | no |
-| use\_host\_volume | Use nomad host volume | bool | false | no |
+| nomad_datacenters | Nomad data centers | list(string) | ["dc1"] | no |
+| nomad_namespace | [Enterprise] Nomad namespace | string | "default" | no |
+| nomad_host_volume | Nomad host volume name | string | "persistence" | no |
+| service_name | Postgres service name | string | "postgres" | no |
+| container_port | Postgres port | number | 5432 | no |
+| container_image | Postgres docker image | string | "postgres:12-alpine" | no |
+| admin_user | Postgres admin username | string | "postgres" | no |
+| admin_password | Postgres admin password | string | "postgres" | no |
+| database | Postgres database name | string | "metastore" | no |
+| container_environment_variables | Postgres container environement variables | list(string) | ["PGDATA=/var/lib/postgresql/data"] | no |
+| volume_destination | Postgres volume destination | string | "/var/lib/postgresql/data" | no |
+| use_host_volume | Use nomad host volume | bool | false | no |
+| use_canary | Switch to use canary deployment for Postgres | bool | no |
+| vault_secret.use_vault_kv | Set if want to access secrets from Vault | bool | false |
+| vault_secret.vault_kv_path | Path to the secret key in Vault | string | "secret/postgres" |
+| vault_secret.vault_kv_username_name | Secret key name in Vault kv path | string | "username" |
+| vault_secret.vault_kv_password_name | Secret key name in Vault kv path | string | "password" |
 
 ## Outputs
 | Name | Description | Type |
 |------|-------------|------|
-| service\_name | Postgres service name | string |
+| service_name | Postgres service name | string |
 | username | Postgres username | string |
 | password | Postgres password | string |
-| database\_name | Postgres database name | string |
+| database_name | Postgres database name | string |
 | port | Postgres port | number |
 
+## Secrets & credentials
+This module presents two ways of setting credentials (username and password). You can set them manually or upload secrets to Vault.
+
+### Set credentials manually
+This module uses the `admin_user` and `admin_password` credentials as default, and contain the value `postgres`.  
+If you want to overwrite the default values, you need to is set the `vault_secret.use_vault_kv` to `false` and leave the rest of the values as blanks. 
```

Comment:
I am a bit lost in this sentence. Do you mean override default values of `admin_user` and `admin_password`?
```suggestion
If you want to overwrite the default values, you need to set the `vault_secret.use_vault_kv` to `false` and leave the rest of the values as blanks. 
```

Reply:
Yes, that's right. It's only ment to indicate that you can overwrite the default values that is set to `postgres`. Do you want me to rephrase the sentence?

Reply:
Yes